src/app/public/src/modules/wrapper/wrapper.component.ts

```diff
@@ -61,25 +61,35 @@ export class StacheWrapperComponent implements OnInit, OnDestroy, AfterViewInit
   @Input()
   public showInNav: boolean = true;
 
+  @Input()
+  public inPageRoutes: StacheNavLink[];
+
   public jsonData: any;
-  public inPageRoutes: StacheNavLink[] = [];
-  private pageAnchorSubscription: Subscription;
+  private ngUnsubscribe = new Subject();
 
   public constructor(
     private config: StacheConfigService,
     private dataService: StacheJsonDataService,
+    private pageAnchorService: StachePageAnchorService,
     private titleService: StacheTitleService,
     private route: ActivatedRoute,
     private navService: StacheNavService,
-    private anchorService: StachePageAnchorService,
     private cdr: ChangeDetectorRef,
     private windowRef: StacheWindowRef,
     private omnibarService: StacheOmnibarAdapterService) { }
 
   public ngOnInit(): void {
     this.omnibarService.checkForOmnibar();
     this.jsonData = this.dataService.getAll();
-    this.registerPageAnchors();
+    if (!this.inPageRoutes) {
+      this.pageAnchorService
+        .anchorStream
+        .takeUntil(this.ngUnsubscribe)
+        .subscribe((anchors: BehaviorSubject<StacheNavLink>[]) => {
+        this.inPageRoutes = anchors
+          .map((anchor: BehaviorSubject<StacheNavLink>) => anchor.getValue());
```

Comment:
I think this is the final change,  we need to add a `this.cdr.detectChanges();` after we update the `inPageRoutes`, so the Wrapper component is aware of the change to this field and can pass it down to the children.